sdk/core/allocator/main.cc

```diff
@@ -215,15 +215,9 @@ namespace
 	{
 		return -EINVAL;
 	}
-	/*
-	 * Since we use the shadow bits to detect valid frees, we need to consult
-	 * the revoker on whether the user cap is valid.
-	 */
-	if (!revoker.is_free_cap_valid(mem))
-	{
-		return -EINVAL;
-	}
-	gm->mspace_free(mem);
+	int rv = gm->mspace_free(mem);
+	if (rv)
```

Comment:
Clang-tidy should complain here.

Reply:
It seems to be okay with it for some reason, but I've fixed the style.